downloader/src/main/kotlin/VersionControlSystem.kt

```diff
@@ -205,10 +205,19 @@ abstract class VersionControlSystem {
         return VcsHost.parseUrl(vcsUrl).type == type || isApplicableUrlInternal(vcsUrl)
     }
 
+    /**
+     * Return a reference to a [CommandLineTool] associated with this concrete [VersionControlSystem] if such a
+     * tool exists. This is used for instance to check whether this [VersionControlSystem] is available. This
+     * default implementation checks whether this class implements the [CommandLineTool] interface and if so, returns
+     * a reference to itself. Subclasses making use of a command line tool without implementing this interface should
+     * override this function.
+     */
+    open fun getCommandLineTool(): CommandLineTool? = this as? CommandLineTool
```

Comment:
There is not really need for exposing the command line tool again, I'd prefer to keep it hidden and thus use constructor injection instead. Would that make sense to you @sschuberth  @oheger-bosch ?

Reply:
I don't have a strong opinion here.

Reply:
@fviernau, how would you then implement the check whether the tool is on the path?

Reply:
In the constructor: `private val commandLineTool: CommandLineTool? = null`
And the check: `fun isAvailable(): Boolean = commandLineTool?.isInPath() ?: true`

Reply:
Ah, I think I got it now. Changed accordingly.

Reply:
Thanks!